[Bug]: ssh:// git repository URLs with userinfo are not recognized as code-hosting repos

### Bug Description

OpenViking documents and routes `ssh://` repository sources as supported git repository URLs, but the code-hosting URL helper compares `urlparse(url).netloc` directly to configured domains. For common clone URLs such as `ssh://git@github.com/org/repo.git`, `netloc` is `git@github.com`, so the URL is not recognized as a configured GitHub/GitLab/code-hosting domain.

This breaks repository metadata detection and helper behavior for a valid supported clone URL shape. For example, `parse_code_hosting_url("ssh://git@github.com/org/repo.git")` returns `None`, and `is_code_hosting_url(...)` / `is_git_repo_url(...)` return `False` even though the hostname is `github.com`.

### Steps to Reproduce

1. Configure the default code-hosting domains, including `github.com`.
2. Call `parse_code_hosting_url("ssh://git@github.com/org/repo.git")`.
3. Call `is_code_hosting_url("ssh://git@github.com/org/repo.git")`.
4. Call `is_git_repo_url("ssh://git@github.com/org/repo.git")`.

### Expected Behavior

The helpers should use the parsed hostname, not raw netloc, so `ssh://git@github.com/org/repo.git` is treated the same as other GitHub repository clone URLs:

- `parse_code_hosting_url(...)` returns `org/repo`
- `is_code_hosting_url(...)` returns `True`
- `is_git_repo_url(...)` returns `True`

### Actual Behavior

The helpers compare `git@github.com` to `github.com` and reject the URL:

- `parse_code_hosting_url(...)` returns `None`
- `is_code_hosting_url(...)` returns `False`
- `is_git_repo_url(...)` returns `False`

### Minimal Reproducible Example

```python
from types import SimpleNamespace
from unittest.mock import patch

from openviking.utils import code_hosting_utils as u

config = SimpleNamespace(
    code=SimpleNamespace(
        github_domains=["github.com"],
        gitlab_domains=["gitlab.com"],
        code_hosting_domains=["github.com", "gitlab.com"],
    )
)

with patch.object(u, "get_openviking_config", return_value=config):
    assert u.parse_code_hosting_url("ssh://git@github.com/org/repo.git") == "org/repo"
    assert u.is_code_hosting_url("ssh://git@github.com/org/repo.git") is True
    assert u.is_git_repo_url("ssh://git@github.com/org/repo.git") is True
```

### Error Logs

```shell
AssertionError: assert None == 'org/repo'
AssertionError: assert False is True
```

### OpenViking Version

main at f9867be

### Python Version

3.12.3

### Operating System

Linux

### Model Backend

Other

### Additional Context

The same raw-netloc comparison also rejects URL forms with explicit ports, such as `https://github.com:443/org/repo`, even though the hostname is still configured. Using `urlparse(...).hostname` makes domain matching consistent across userinfo and port-bearing URLs.

--- a/openviking/utils/code_hosting_utils.py
+++ b/openviking/utils/code_hosting_utils.py
@@ -13,6 +13,18 @@ from urllib.parse import urlparse
 from openviking_cli.utils.config import get_openviking_config
 
 
+def _extract_host(url: str) -> str:
+    """Extract normalized host for supported git/code-hosting URL forms."""
+    if url.startswith("git@"):
+        rest = url[4:]
+        if ":" not in rest:
+            return ""
+        return rest.split(":", 1)[0].strip().lower()
+
+    parsed = urlparse(url)
+    return (parsed.hostname or parsed.netloc or "").strip().lower()
+
+
 def parse_code_hosting_url(url: str) -> Optional[str]:
     """Parse code hosting platform URL to get org/repo path.
 
@@ -32,6 +44,7 @@ def parse_code_hosting_url(url: str) -> Optional[str]:
             + config.code.code_hosting_domains
         )
     )
+    host = _extract_host(url)
 
     # Handle git@ SSH URLs: git@host:org/repo.git
     if url.startswith("git@"):
@@ -60,7 +73,7 @@ def parse_code_hosting_url(url: str) -> Optional[str]:
 
     # For GitHub/GitLab URLs with org/repo structure
     if (
-        parsed.netloc in config.code.github_domains + config.code.gitlab_domains
+        host in config.code.github_domains + config.code.gitlab_domains
         and len(path_parts) >= 2
     ):
         # Take first two parts: org/repo
@@ -86,7 +99,7 @@ def is_github_url(url: str) -> bool:
         True if the URL is a GitHub URL
     """
     config = get_openviking_config()
-    return urlparse(url).netloc in config.code.github_domains
+    return _extract_host(url) in config.code.github_domains
 
 
 def is_gitlab_url(url: str) -> bool:
@@ -99,7 +112,7 @@ def is_gitlab_url(url: str) -> bool:
         True if the URL is a GitLab URL
     """
     config = get_openviking_config()
-    return urlparse(url).netloc in config.code.gitlab_domains
+    return _extract_host(url) in config.code.gitlab_domains
 
 
 def is_code_hosting_url(url: str) -> bool:
@@ -127,7 +140,7 @@ def is_code_hosting_url(url: str) -> bool:
         host_part = url[4:].split(":", 1)[0]
         return host_part in all_domains
 
-    return urlparse(url).netloc in all_domains
+    return _extract_host(url) in all_domains
 
 
 def validate_git_ssh_uri(url: str) -> None:
@@ -173,7 +186,7 @@ def is_git_repo_url(url: str) -> bool:
             )
         )
         parsed = urlparse(url)
-        if parsed.netloc not in all_domains:
+        if _extract_host(url) not in all_domains:
             return False
         path_parts = [p for p in parsed.path.split("/") if p]
         # Strip .git suffix from last part for counting

--- a/tests/test_code_hosting_utils.py
+++ b/tests/test_code_hosting_utils.py
@@ -81,6 +81,14 @@ def test_parse_code_hosting_url_https_dotgit():
     assert parse_code_hosting_url("https://github.com/org/repo.git") == "org/repo"
 
 
+def test_parse_code_hosting_url_ssh_with_userinfo():
+    assert parse_code_hosting_url("ssh://git@github.com/org/repo.git") == "org/repo"
+
+
+def test_parse_code_hosting_url_https_with_explicit_port():
+    assert parse_code_hosting_url("https://github.com:443/org/repo.git") == "org/repo"
+
+
 # --- validate_git_ssh_uri ---
 
 
@@ -118,6 +126,14 @@ def test_is_code_hosting_url_https():
     assert is_code_hosting_url("https://github.com/org/repo") is True
 
 
+def test_is_code_hosting_url_ssh_with_userinfo():
+    assert is_code_hosting_url("ssh://git@github.com/org/repo.git") is True
+
+
+def test_is_code_hosting_url_https_with_explicit_port():
+    assert is_code_hosting_url("https://github.com:443/org/repo") is True
+
+
 # --- is_git_repo_url ---
 
 
@@ -127,6 +143,14 @@ def test_is_git_repo_url_git_ssh():
 
 def test_is_git_repo_url_https_repo():
     assert is_git_repo_url("https://github.com/org/repo") is True
+
+
+def test_is_git_repo_url_ssh_with_userinfo():
+    assert is_git_repo_url("ssh://git@github.com/org/repo.git") is True
+
+
+def test_is_git_repo_url_https_with_explicit_port():
+    assert is_git_repo_url("https://github.com:443/org/repo") is True
 
 
 def test_is_git_repo_url_https_issues():

